libutils/file_lib.c

```diff
@@ -736,14 +725,27 @@ int safe_chmod(const char *path, mode_t mode)
 #ifdef __MINGW32__
     return chmod(path, mode);
 #else // !__MINGW32__
-    int fd = safe_open(path, 0);
-    if (fd < 0)
-    {
-        return -1;
-    }
+    int dirfd = -1;
+    int ret = -1;
+
+    char *parent_dir_alloc = xstrdup(path);
+    char *leaf_alloc = xstrdup(path);
+    char *parent_dir = dirname(parent_dir_alloc);
+    char *leaf = basename(leaf_alloc);
+
+    if ((dirfd = safe_open(parent_dir, O_RDONLY)) == -1)
+        goto bad;
+
+    if ((ret = fchmodat(dirfd, leaf, mode, AT_SYMLINK_NOFOLLOW)))
```

Comment:
``flags`` should be ``0``, not ``AT_SYMLINK_NOFOLLOW``, since this is ``chmod``, not ``lchmod``.